src/testUtils/index.tsx

```diff
@@ -0,0 +1,42 @@
+import { mount, shallow } from 'enzyme';
+import * as React from 'react';
+
+export interface Resp {
+    results: Array<{ [key: string]: any }>;
+    total?: number;
+}
+
+export interface QueryString {
+    [key: string]: string;
+}
+
+/**
+ * Compose setup method for component tests
+ */
+export const composeSetup = <P extends {}, C extends {} = { [contextType: string]: any }>(
+    defaultProps: P,
+    context: C | Partial<C>,
+    Component: React.ComponentClass | React.SFC
+) => (propOverrides: P | Partial<P> = {}, shallowRender: boolean = false) => {
+    // Waiting on https://github.com/Microsoft/TypeScript/pull/1328
+    const props = Object.assign({}, defaultProps, propOverrides);
+
+    // prettier-ignore
+    const wrapper = (
+        // tslint:disable-next-line:ban-types
+        shallowRender ? (shallow as Function) : (mount as Function)
+    )(
+        <Component {...props} />, { context }
+    );
+
+    return {
+        wrapper,
+        props,
+        context
+    };
+};
+
+/**
+ * Wait for promises in queue to resolve
+ */
+export const yieldToPromises = () => new Promise(resolve => setImmediate(resolve));
```

Comment:
helper to call before making assertions on functionality that depends on async work, e.g. 

```
const { wrapper } = setup();

// assuming a click event initiates a network request
getSpecWrapper(wrapper, 'load-content').simulate('click');

// wait for promise to resolve
await yieldToPromises();

// force a re-render
wrapper.update();

expect(getSpecWrapper(wrapper, 'loaded-content')).toBe(...);

```

Reply:
i saw this elsewhere but the use of the term 'flush' was a bit confusing for me; saw this as more of a deferral. happy to use if it's a common thing 👍